numpy/core/arrayprint.py

```diff
@@ -616,14 +616,14 @@ def array2string(a, max_line_width=None, precision=None,
 
 
 def _extendLine(s, line, word, max_line_len, next_line_prefix):
-    if len(line.rstrip()) + len(word.rstrip()) >= max_line_len:
+    if len((line + word).rstrip()) > max_line_len:
```

Comment:
Stripping `line` made no sense here. We remain compatible by also fixing the comparison, since in practice `line` had exactly one trailing space (`<snip>, '` or `<snip> '`), except when a custom separator was passed.